demo/dismissable/dismissable-item.js

```diff
@@ -0,0 +1,218 @@
+const kMinFlingVelocityValue = 0.4;
+const kTouchSlopValue = 5;
+
+class VelocityTracker {
+  constructor() {
+    this._recentTouchMoves = [];
+    this._timeWindow = 50;
+  }
+
+  _pruneHistory(timeStamp) {
+    const index = this._recentTouchMoves.findIndex((touch) => {
+      return touch.timeStamp > timeStamp - this._timeWindow;
+    });
+    this._recentTouchMoves.splice(0, index + 1);
+  }
+
+  update(e) {
+    this._pruneHistory(e.timeStamp);
+    this._recentTouchMoves.push(e);
+
+    const oldestTouchMove = this._recentTouchMoves[0];
+
+    const deltaX = e.clientX - oldestTouchMove.clientX;
+    const deltaT = e.timeStamp - oldestTouchMove.timeStamp;
+
+    return {
+      velocityX: (deltaT > 0) ? deltaX / deltaT : 0
+    };
+  }
+}
+
+class DismissableItem extends HTMLElement {
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
```

Comment:
seems like we don't really need a shadowRoot for this element

Reply:
Right, I was thinking the same earlier. Done